### PR TITLE
Display all challenges with description without login.

### LINKED ISSUE #1247: Challenge List page should not require login
What the title says. 

Just like the VQA challenge details are not being auth-wall, everything else should be visible as well.

### PR DESCRIPTION
**Pull request by RishabhJain2018**
Fixes #1247. @deshraj @aka-jain @spyshiv Please review the PR.

### CODE REVIEW

**Comment by deshraj** [comment]
@RishabhJain2018: can you please share the screenshot?

File: frontend/src/js/controllers/challengeListAllCtrl.js [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    'use strict';
+    angular
+        .module('evalai')
+        .controller('ChallengeListAllCtrl', ChallengeListAllCtrl);
+
+    ChallengeListAllCtrl.$inject = ['utilities'];
+
+    function ChallengeListAllCtrl(utilities) {
+        var vm = this;
+        // vm.challengeId = $stateParams.challengeId;
```
Comment (spyshiv) [inline frontend/src/js/controllers/challengeListAllCtrl.js]: @RishabhJain2018: Please remove this comment.

File: frontend/src/js/controllers/challengeListAllCtrl.js [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+        // get details of the particular challenge
+        var parameters = {};
+        parameters.url = 'challenges/challenge/' + challengeId + '/';
+        parameters.method = 'GET';
+        parameters.data = {};
+        parameters.callback = {
+            onSuccess: function(response) {
+                var details = response.data;
+                console.log("abcccc");
```
Comment (spyshiv) [inline frontend/src/js/controllers/challengeListAllCtrl.js]: Remove these consoles here

File: frontend/src/views/web/challenge/list_all_challenges.html [hunk 3]
```diff
@@ -1,92 +1,118 @@
...
+        </div>
         </div>
-    </div>
-    <br>
-</section>
+        <br>
+    </section>
     <landing-footer></landing-footer>
-</div>
+</div>
```
Comment (aka-jain) [inline frontend/src/views/web/challenge/list_all_challenges.html]: New line

File: frontend/src/js/route-config/route-config.js [hunk 4]
```diff
@@ -518,20 +591,32 @@
...
-                $state.transitionTo("home");
-                event.preventDefault();
-                return false;
-            } else if (utilities.isAuthenticated()) {
+            var path = $location.path(); // get the current path
+            var pathArray = path.split('/'); // split it to make a array
+            var challengeId = pathArray[pathArray.length-2]; // get the challenge id
+
+            if (toState.url === '/leaderboard' || toState.url === '/overview'
+                || toState.url === '/evaluation' || toState.url === '/phases'
```
Comment (aka-jain) [inline frontend/src/js/route-config/route-config.js]: Instead of using these conditions just take challenge as reference.

Comment (deshraj) [reply frontend/src/js/route-config/route-config.js]: I am going with these conditions for now. @RishabhJain2018 can you please incorporate @aka-jain's commit in a separate pull request?

**Pull request merged**